src/components/ChecTagsField.vue

```diff
@@ -0,0 +1,368 @@
+          @blur="handleInputBlur"
+        >
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+import ChecTag from './ChecTag';
+
+export default {
+  name: 'ChecTagsField',
+  components: {
+    ChecTag,
+  },
+  props: {
+    /**
+     * If the textfield holds a value and loses focus,
+     * a tag will be added out of the value
+     */
+    addOnBlur: Boolean,
+    /**
+     * Add a tag when listed keys are pressed
+     * 13 (return), 188 (comma), 9(tab)
+     */
+    addOnKeys: {
+      type: Array,
+      default: () => [
+        13, // Return
+        188, // Comma ','
+        9, // Tab
+      ],
+    },
+    /**
+     * Allow same tags to be added more than once
+     */
+    allowDuplicates: Boolean,
+    /**
+     * Function to pass before an option is added
+     * Ie. Validation rules
+     */
+    beforeAdding: {
+      type: Function,
+      default: (tag) => tag,
+    },
+    /**
+     * The maximum amount of characters the input/tags is allowed to have
+     */
+    maxLength: {
+      type: Number,
+      default: 20,
+    },
+    /**
+     * The label for the tags field input
+     */
+    placeholder: {
+      type: String,
+      default: '',
+    },
+    /**
+     * Options list for the tags added
+     */
+    optionsList: {
+      type: Array,
+      default: () => [],
+    },
+    /**
+     * Controls the width of the tags field. Use one of the tailwind sizes, e.g. 64, 72, 80, 88, 96.
+     */
+    width: {
+      type: String,
+      default: '88',
+    },
+  },
+  data() {
+    return {
+      newTag: '',
+      options: [...this.optionsList],
+      activeInput: false,
+      activeTag: false,
+      isInputVisible: true,
+      tagsFieldFocused: false,
+    };
+  },
+  computed: {
+    classNames() {
+      const { tagsFieldFocused, width } = this;
+      // Bind an active modifier to the tags field element
+      return [
+        'tags-field',
+        `w-${width}`,
+        `min-w-${width}`,
+        {
+          'tags-field--active': tagsFieldFocused,
+        },
+      ];
+    },
+  },
+  watch: {
+    /**
+     * Watch for new tags being added
+     */
+    optionsList() {
+      this.options = [...this.optionsList];
+    },
+  },
+  created() {
+    // Add event listener to listen to outside click events
+    window.addEventListener('click', this.onOutsideClick);
+
+    // When 'Escape' key is pressed
+    const onEscape = (e) => {
+      if (e.key === 'Escape') {
+        // Blur the tags field element
+        this.tagsFieldFocused = false;
+        // Set input to not visible
+        this.handleInputVisibility();
+        // Ensure that if tags field is not focused
+        // the input is not active
+        if (!this.tagsFieldFocused) {
+          this.activeInput = false;
+        }
+      }
+    };
+    document.addEventListener('keydown', onEscape);
+    this.$once('hook:destroyed', () => {
+      document.removeEventListener('keydown', onEscape);
+    });
```

Comment:
What's this? I haven't seen this sort of thing before. The `onOutsideClick` is the more standard way of doing this.